### PR TITLE
Upgrade packages for PHP 7.2 and adjust the ruleset for BC breaks.

### PR DESCRIPTION
**Pull request by frankverhoeven**
This allows services that do not rely on True for their hosting to use
the latest packages.
This PR includes additions to the ruleset that weren't available before,
see the changes for details.

Proposed version for release: `1.1.0`. This allows a simple `composer update` to upgrade as soon as other packages are upgraded for 7.2, instead of having to switch to a new 2.0 version. For current usages this does not impose a problem due to the php constraint.

### CODE REVIEW

File: MyOnlineStore/ruleset.xml [hunk 1]
```diff
@@ -168,12 +167,12 @@
     <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
         <properties>
             <property name="newlinesCountBetweenOpenTagAndDeclare" value="1"/>
-            <property name="newlinesCountAfterDeclare" value="1"/>
+            <property name="newlinesCountAfterDeclare" value="2"/>
```
Comment (Feijs) [inline MyOnlineStore/ruleset.xml:170]: Why this extra empty line requirement?

Comment (frankverhoeven) [reply MyOnlineStore/ruleset.xml:170]: Two newlines = one empty line. It was wrong before, but not used yet due to the older version.

**Review by Feijs — APPROVED** [review]

**Comment by Feijs** [comment]
> Proposed version for release: 1.1.0. This allows a simple composer update to upgrade as soon as other packages are upgraded for 7.2, instead of having to switch to a new 2.0 version. For current usages this does not impose a problem due to the php constraint.

Sure, though I would propose to already create a branch (`php7.0` , or `1.0.x`) to promote future changes being applied to both minor versions

**Comment by frankverhoeven** [comment]
Sounds good to me.

Added a `1.0.x` branch.

**Pull request merged**